src/clients/java/ci.zig

```diff
@@ -0,0 +1,60 @@
+const std = @import("std");
+const log = std.log;
+const assert = std.debug.assert;
+
+const flags = @import("../../flags.zig");
+const fatal = flags.fatal;
+const Shell = @import("../../shell.zig");
+const TmpTigerBeetle = @import("../../testing/tmp_tigerbeetle.zig");
+
+pub fn tests(shell: *Shell, gpa: std.mem.Allocator) !void {
+    // Java's maven doesn't support a separate test command, or a way to add dependency on a
+    // project (as opposed to a compiled jar file).
+    //
+    // For this reason, we do all our testing in one go, imperatively building a client jar and
+    // installing it into a temporary maven repository.
+    try shell.exec("mvn --batch-mode --file pom.xml --quiet formatter:validate", .{});
+    try shell.exec("mvn --batch-mode --file pom.xml --quiet package", .{});
+
+    var tmp_dir = std.testing.tmpDir(.{});
+    defer tmp_dir.cleanup();
+
+    const maven_repo_path = try tmp_dir.dir.realpathAlloc(shell.arena.allocator(), ".");
+
+    try shell.exec(
+        \\mvn deploy:deploy-file --quiet
+        \\  -Durl=file://{maven_repo_path}
+        \\  -Dfile=target/tigerbeetle-java-0.0.1-SNAPSHOT.jar
+        \\  -DgroupId=com.tigerbeetle -DartifactId=tigerbeetle-java -Dversion=0.0.0
+        \\  -Dpackaging=jar
+    , .{
+        .maven_repo_path = maven_repo_path,
+    });
```

Comment:
Work-around for the lack of path deps in maven.

Reply:
I know it's a placeholder, but for `target/tigerbeetle-java-0.0.1-SNAPSHOT.jar` should we not use `target/tigerbeetle-java-0.0.0-SNAPSHOT.jar`?

Reply:
Figured out how to solve this in a better way, with less version patching!